sarcophagus-org/sarcophagus-v2-app
# 88 webapp arch node 2nd handshake

## Overview
Should be merged with [this PR](https://github.com/sarcophagus-org/sarcophagus-v2-archaeologist-service/pull/46) on the service

This is a fairly chunky PR. Kinda got over-eager in getting as close to "create sarco" as possible.

### To test:
- Have fully registered arch(s) running
- Go through the ff steps, in-app, of sarco creation:
   - Name sarco, set resurrection time
   - upload a file
   - set sharding thresholds
   - select archaeologists
   - click the 'next page' button under the archaeologists table
- The handshake and negotiation process should continue automated from there. Console logs should indicate this happening.
- Modify `negotiationParams` properties in `useSarcophagusNegotiation` to test different error responses.

--- 

### Some things to note:

- Sarcophagus creation is currently failing contract side due to signature mismatch. We gotta look into why this is happening -- signature creation is using the exact same code that the create sarco tests are using.

- `handleSubmit` hook renamed to `uploadToArweave`, as it seemed that's really all it was doing under the hood. I got the impression from a recent standup that it may or may not be removed later anyway. In any case I'd suggest keeping a similar name wherever it ends up finding its home.

- `dialSelectedArchaeologists` was moved to a new `useSarcophagusNegotiation` hook.

- `setPayloadTxId`, `setShardsTxId`, and `setShards` actions were combined into a `setShardPayloadData` action. This is residue from attempts to solve a state bug and not terribly necessary in the grand scheme of things, but I left it that way because a single dispatch call reads better anyway.

- `useCreateSarcohpagus`'s shard setup and encryption now tie shards with archaeologists in a single, easy to reference `ArchaeologistEncryptedShard` object.

- `recipientState.publicKey` is being skipped entirely so that signature negotiation can be tested more easily. These should be restored in a later PR. Added TODOs wherever I'm doing this.

- I had to update `updateArchProperty` to account for updates to `selectedArchaeologists` updated properties as well. We need to think through the pros and cons of removing `selectedArchaeologists` and using and `isSelected` flag instead on the `archaeologists` state.
